### PR TITLE
Implement fixtures

### LINKED ISSUE #57: Be able to set default value on the channel
It could be nice to set default values on the channel scope.

```yaml
monsieurbiz_sylius_settings:
    plugins:
        app.my_plugin:
            vendor_name: Monsieur Biz
            vendor_url:
            plugin_name: My Plugin
            description: My amazing plugin
            icon: rocket
            use_locales: false
            classes:
                form: App\Form\Type\SettingsType
            default_values:
                value_1: 'value_1'
                value_2: 'value_2'
            MY_CHANNEL_CODE_values:
                value_1: 'value_1_1'
                value_2: 'value_2_2'
```

### PR DESCRIPTION
**Pull request by jacquesbh**
We just implemented the fixtures 🔥.

fixes #57

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -63,14 +63,18 @@
...
         }
     },
     "config": {
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true,
             "symfony/thanks": true,
             "ergebnis/composer-normalize": true,
-            "symfony/flex": true
+            "symfony/flex": true,
+            "php-http/discovery": false
```
Comment (jacquesbh) [inline composer.json:77]:
```suggestion
            "php-http/discovery": true
```

File: src/Command/SetSettingsCommand.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            $locale = $input->getOption(self::OPTION_LOCALE);
+
+            $channel = null;
+            if (null !== $channelCode) {
+                /** @var ?ChannelInterface $channel */
+                $channel = $this->channelRepository->findOneBy(['code' => $channelCode]);
+            }
+
+            /** @var SettingsInterface $settings */
+            $settings = $this->settingsRegistry->getByAlias($alias);
```
Comment (jacquesbh) [inline src/Command/SetSettingsCommand.php:110]: **🎯 suggestion:** ‏Can be null right? So maybe we could add a test here and throw an error if we don't find the Settings by its alias.

File: src/Command/SetSettingsCommand.php [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            $setting = $this->settingFactory->createNew();
+            $setting->setVendor($vendor);
+            $setting->setPlugin($plugin);
+            $setting->setPath($path);
+            $setting->setLocaleCode($locale);
+            $setting->setChannel($channel);
+        }
+
+        return $setting;
+    }
```
Comment (jacquesbh) [inline src/Command/SetSettingsCommand.php:164]: **🎯 suggestion:** ‏This code seems to be a duplicate of the Fixtures, maybe we could have a specific service for it and use it in both places?

File: src/Command/SetSettingsCommand.php [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            SettingInterface::STORAGE_TYPE_FLOAT,
+            SettingInterface::STORAGE_TYPE_DATETIME,
+            SettingInterface::STORAGE_TYPE_DATE,
+            SettingInterface::STORAGE_TYPE_JSON,
+        ];
+
+        if (!\in_array($type, $types, true)) {
+            throw new Exception(sprintf('The type "%s" is not valid. Valid types are: %s', $type, implode(', ', $types)));
+        }
+    }
```
Comment (jacquesbh) [inline src/Command/SetSettingsCommand.php:181]: **🎯 suggestion:** ‏This part could be in the same service as mentioned above.

File: src/Fixture/Factory/SettingsFixtureFactory.php [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            ->setDefault('path', '')
+            ->setAllowedTypes('path', 'string')
+            ->setDefault('channel', null)
+            ->setAllowedTypes('channel', ['null', 'string'])
+            ->setDefault('locale', null)
+            ->setAllowedTypes('locale', ['null', 'string'])
+            ->setDefault('type', SettingInterface::STORAGE_TYPE_TEXT)
+            ->setAllowedTypes('type', 'string')
+            ->setAllowedValues('type', [SettingInterface::STORAGE_TYPE_TEXT, SettingInterface::STORAGE_TYPE_BOOLEAN, SettingInterface::STORAGE_TYPE_INTEGER,
+                SettingInterface::STORAGE_TYPE_FLOAT, SettingInterface::STORAGE_TYPE_JSON, SettingInterface::STORAGE_TYPE_DATE, SettingInterface::STORAGE_TYPE_DATETIME])
```
Comment (jacquesbh) [inline src/Fixture/Factory/SettingsFixtureFactory.php:113]: **🎯 suggestion:** We could have a static method into the SettingInterface implementation returning all the types? Like `Setting::getAllStorageTypes():array`.‏

File: src/Formatter/SettingsFormatter.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusSettingsPlugin\Formatter;
+
+use DateTime;
+use MonsieurBiz\SyliusSettingsPlugin\Entity\Setting\SettingInterface;
+
+class SettingsFormatter implements SettingsFormatterInterface
```
Comment (jacquesbh) [inline src/Formatter/SettingsFormatter.php:19]: Love it! 😍

File: README.md [hunk 7]
```diff
@@ -101,6 +101,34 @@ use MonsieurBiz\SyliusSettingsPlugin\Provider\SettingsProviderInterface;
...
+You can use a CLI command to set a value for a setting directly from the console:
+`$ ./bin/console monsieurbiz:settings:set {alias} {path} {type} {value} --channel="FASHION_WEB" --locale="en_US"`
+
+Examples:
+```bash
+$ ./bin/console monsieurbiz:settings:set app.default demo_message text 'fashion message' --channel="FASHION_WEB" --locale="en_US"
+$ ./bin/console monsieurbiz:settings:set app.default demo_message json '{"foo":"baz"}' --channel="FASHION_WEB" --locale="en_US"
+$ ./bin/console monsieurbiz:settings:set app.default demo_message datetime '2023-07-24 01:02:03' --channel="FASHION_WEB" --locale="en_US"
+$ ./bin/console monsieurbiz:settings:set app.default enabled boolean 0
+```
```
Comment (jacquesbh) [inline README.md:128]: **❔ question:** ‏If the type of the setting is already set, with an existing setting as example, could we omit it as well?

**Review by jacquesbh — COMMENTED** [review]
🔥🔥🔥

**Review by maximehuran — APPROVED** [review]

**Review by etienne-monsieurbiz — APPROVED** [review]

**Pull request merged**